### PR TITLE
Standard config - Add new e2e tests for the cht-form web component

### PR DESCRIPTION
**Pull request by tatilepizs**
# Description

This PR adds new automated tests for the cht-form web component. It covers all the available forms for the `standard` config:

- [x] child_health_registration
- [x] death_confirmation
- [x] delivery
- [x] immunization_visit
- [x] postnatal_visit
- [x] pregnancy
- [x] pregnancy_visit
- [x] health_center-create
- [x] health_center-edit

medic/cht-core#[number]

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs

If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:cht-form-e2e-tests-standard/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:cht-form-e2e-tests-standard/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:cht-form-e2e-tests-standard/docker-compose/cht-couchdb-clustered.yml)

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: tests/page-objects/standard/enketo/pregnancy.wdio.page.js [hunk 1]
```diff
@@ -4,19 +4,26 @@ const enketoCommonPage = require('@page-objects/standard/enketo/enketo.wdio.page
...
+const patientIdSummary = () => $(FORM +
+  ` span[data-itext-id="/pregnancy/group_review/r_pregnancy_details:label"]${enketoCommonPage.ACTIVE} ` +
+  enketoCommonPage.patientIdSummary('pregnancy', 'review'));
+const riskFactorsSummary = () => $$(`${FORM} :not(label.disabled):not(label.or-appearance-yellow)  > ` +
   `span[data-itext-id*="/pregnancy/group_review/r_risk_factor"]${enketoCommonPage.ACTIVE}`);
-const dangerSignsSummary = () => $$(`${form} span[data-itext-id*="/pregnancy/group_review/r_danger_sign"]` +
-  `${enketoCommonPage.ACTIVE}`);
-const followUpSMS = () => $(`${form} ${enketoCommonPage.followUpSms('pregnancy')}`);
+const dangerSignsSummary = () => $$(`${FORM} span[data-itext-id*="/pregnancy/group_review/r_danger_sign"]` +
+  enketoCommonPage.ACTIVE);
```
Comment (jkuester) [inline tests/page-objects/standard/enketo/pregnancy.wdio.page.js:24]:
```suggestion
const { ACTIVE } = enketoCommonPage;
const patientNameSummary = () => {
  const nameSum = enketoCommonPage.patientNameSummary('pregnancy');
  return $(`${FORM} span[data-itext-id="/pregnancy/group_review/r_pregnancy_details:label"]${ACTIVE} ${nameSum}`);
};
const patientIdSummary = () => {
  const idSum = enketoCommonPage.patientIdSummary('pregnancy', 'review');
  return $(`${FORM} span[data-itext-id="/pregnancy/group_review/r_pregnancy_details:label"]${ACTIVE} ${idSum}`);
};
const riskFactorsSummary = () => {
  const parent = ':not(label.disabled):not(label.or-appearance-yellow)';
  return $$(`${FORM} ${parent}  > span[data-itext-id*="/pregnancy/group_review/r_risk_factor"]${ACTIVE}`);
};
const dangerSignsSummary = () => {
  return $$(`${FORM} span[data-itext-id*="/pregnancy/group_review/r_danger_sign"]${ACTIVE}`);
};
```

File: tests/page-objects/standard/enketo/pregnancy-visit.wdio.page.js [hunk 2]
```diff
@@ -5,9 +5,16 @@ const enketoCommonPage = require('@page-objects/standard/enketo/enketo.wdio.page
...
-  `${enketoCommonPage.ACTIVE}`);
-const followUpSMS = () => $(`${FORM} ${enketoCommonPage.followUpSms('pregnancy_visit')}`);
+const dangerSignSummary = () => $$(FORM +
+  ` span[data-itext-id*="/pregnancy_visit/group_review/r_danger_sign"]${enketoCommonPage.ACTIVE}`);
+const patientNameSummary = () => $(FORM +
+  ` span[data-itext-id="/pregnancy_visit/group_review/r_pregnancy_details:label"]${enketoCommonPage.ACTIVE} ` +
+  enketoCommonPage.patientNameSummary('pregnancy_visit'));
+const patientIdSummary = () => $(FORM +
+  ` span[data-itext-id="/pregnancy_visit/group_review/r_pregnancy_details:label"]${enketoCommonPage.ACTIVE} ` +
+  enketoCommonPage.patientIdSummary('pregnancy_visit', 'review'));
```
Comment (jkuester) [inline tests/page-objects/standard/enketo/pregnancy-visit.wdio.page.js:15]:
```suggestion
const { ACTIVE } = enketoCommonPage;
const dangerSignSummary = () => {
  return $$(`${FORM} span[data-itext-id*="/pregnancy_visit/group_review/r_danger_sign"]${ACTIVE}`);
};
const patientNameSummary = () => {
  const nameSum = enketoCommonPage.patientNameSummary('pregnancy_visit');
  return $(`${FORM} span[data-itext-id="/pregnancy_visit/group_review/r_pregnancy_details:label"]${ACTIVE} ${nameSum}`);
};
const patientIdSummary = () => {
  const idSum = enketoCommonPage.patientIdSummary('pregnancy_visit', 'review');
  return $(`${FORM} span[data-itext-id="/pregnancy_visit/group_review/r_pregnancy_details:label"]${ACTIVE} ${idSum}`);
};
```

File: tests/e2e/cht-form/standard/delivery.wdio-spec.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
+const mockConfig = require('../mock-config');
+const genericForm = require('@page-objects/default/enketo/generic-form.wdio.page');
+const deliveryForm = require('@page-objects/standard/enketo/delivery.wdio.page');
+const moment = require('moment/moment');
```
Comment (jkuester) [inline tests/e2e/cht-form/standard/delivery.wdio-spec.js:4]:
```suggestion
const moment = require('moment');
```

**Review by jkuester — APPROVED** [review]
So great!  I added some minor suggestions in-line to just avoid those multi-line implicit returns that Gareth mentioned on the default config tests.  Otherwise, all this is awesome!

I say we go ahead and just merge this into the main `7462_forms_module`.  Then Mokhtar can review it all together in that PR next week!

**Comment by tatilepizs** [comment]
Thank you for your detailed review @jkuester. I will merge it in `7462_forms_module` and wait for Mokhtar's review there 😄

**Pull request merged**